Source/ACE/Entity/Player.cs

```diff
@@ -1796,33 +1827,106 @@ public void HandleActionWieldItem(Container container, uint itemId, uint locatio
+                            // Magic numbers - need to understand and fix.
+                            item.ParentLocation = 1;
+                            item.CurrentWieldedLocation = (EquipMask)location;
+                            item.Location = Location;
+                            item.SetPhysicsDescriptionFlag(item);
+                        }
+                        else
+                        {
+                            item.CurrentWieldedLocation = (EquipMask)location;
+                            item.SetPhysicsDescriptionFlag(item);
+                        }
                         UpdateAppearance(container, itemId);
-                        Session.Network.EnqueueSend(
-                            new GameMessageSound(Guid, Sound.WieldObject, (float)1.0),
-                            new GameMessageObjDescEvent(this),
-                            new GameMessageUpdateInstanceId(container.Guid, itemGuid, PropertyInstanceId.Wielder),
-                            new GameEventWieldItem(Session, itemGuid.Full, location));
-                        CurrentLandblock.EnqueueBroadcast(
-                            Location,
-                            Landblock.MaxObjectRange,
-                            new GameMessageObjDescEvent(this));
+                        if ((EquipMask)location == EquipMask.Ammunition)
+                            Session.Network.EnqueueSend(
+                                new GameEventWieldItem(Session, itemGuid.Full, location),
+                                new GameMessageSound(Guid, Sound.WieldObject, (float)1.0));
+                        else
+                        {
+                            if ((EquipMask)location > EquipMask.RingLeft)
+                            {
+                                Session.Network.EnqueueSend(
+                                    new GameMessageCreateObject(item),
+                                    new GameMessageParentEvent(Session.Player, item.Guid, childLocation, placementId),
+                                    new GameEventWieldItem(Session, itemGuid.Full, location),
+                                    new GameMessageSound(Guid, Sound.WieldObject, (float)1.0),
+                                    new GameMessageUpdateInstanceId(container.Guid, new ObjectGuid(0),
+                                        PropertyInstanceId.Container),
+                                    new GameMessageUpdateInstanceId(container.Guid, itemGuid,
+                                        PropertyInstanceId.Wielder),
+                                    new GameMessagePrivateUpdatePropertyInt(Session, PropertyInt.CurrentWieldedLocation,
+                                        location));
+                                CurrentLandblock.EnqueueBroadcast(
+                                    Location,
+                                    Landblock.MaxObjectRange,
+                                    new GameMessageCreateObject(item),
```

Comment:
You've got some duplicated broadcasts here -- e.g. GameMessageCreateObject is sent directly to the client -- then again on broadcast.